### PR TITLE
Implement AsyncOpenSearch() parameter `ssl_assert_hostname`

### LINKED ISSUE #842: [BUG] Async client is missing ssl_assert_hostname implementation
### What is the bug?
The `ssl_assert_hostname` parameter has no effect on the `AsyncOpenSearch` client.

### How can one reproduce the bug?
```python
client = AsyncOpenSearch(hosts='my.opensearch.with.selfsigned.certs', ssl_assert_hostname=False)
await client.search(...)
```
```
opensearchpy.exceptions.ConnectionError: ConnectionError(Cannot connect to host my.opensearch.instance:9200 ssl:True [SSLCertVerificationError: (1, "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'my.opensearch.instance'. (_ssl.c:1000)")]) caused by: ClientConnectorCertificateError(Cannot connect to host my.opensearch.instance:9200 ssl:True [SSLCertVerificationError: (1, "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'my.opensearch.instance'. (_ssl.c:1000)")])
```

### What is the expected behavior?
The connection should succeed even if the hostname does not match.

### What is your host/environment?
N/A

### Do you have any screenshots?
N/A

### Do you have any additional context?
The parameter is documented in the function docs:
https://github.com/opensearch-project/opensearch-py/blob/6fa54a719a399cdcd8d970605487b9e14d257383/opensearchpy/_async/http_aiohttp.py#L120

I believe the parameter works with the `urllib3`-based client.

I'm working on a PR to fix this.

### PR DESCRIPTION
**Pull request by merlinz01**
### Description
Implements the `ssl_assert_hostname` parameter for `AsyncOpenSearch()`.

### Issues Resolved
Closes #842

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — CHANGES_REQUESTED** [review]
I know it's a fairly trivial change, but we want to make sure someone doesn't break this code in the future, so let's add a test, please?

Mention it somewhere in the user guide in this repo, too.

**Comment by dblock** [comment]
Fix DCO, rebase please?

File: test_opensearchpy/test_async/test_connection.py [hunk 1]
```diff
@@ -97,6 +97,12 @@ async def test_ssl_context(self) -> None:
         assert con.use_ssl
         assert con.session.connector._ssl == context
 
+    async def test_ssl_assert_hostname(self) -> None:
```
Comment (dblock) [inline test_opensearchpy/test_async/test_connection.py:100]: Add a test for the default value and a True value, too.

Comment (merlinz01) [reply test_opensearchpy/test_async/test_connection.py:100]: Done.

**Comment by dblock** [comment]
CI failed in lint, `nox -rs format` I think ...

**Review by dblock — APPROVED** [review]

**Review by dblock — CHANGES_REQUESTED** [review]
Looks like a legit test failure with a newer Python version?

```
_ TestAIOHttpConnection.test_nowarn_when_test_uses_https_if_verify_certs_is_off _
self = <test_opensearchpy.test_async.test_connection.TestAIOHttpConnection object at 0x7f8f401a8740>
    async def test_nowarn_when_test_uses_https_if_verify_certs_is_off(self) -> None:
        with warnings.catch_warnings(record=True) as w:
            con = AIOHttpConnection(
                use_ssl=True, verify_certs=False, ssl_show_warn=False
            )
            await con._create_aiohttp_session()
>           assert w == []
E           assert [<warnings.WarningMessage object at 0x7f8f304f7ef0>] == []
E             
E             Left contains one more item: <warnings.WarningMessage object at 0x7f8f304f7ef0>
E             
E             Full diff:
E             - []
E             + [
E             +     <warnings.WarningMessage object at 0x7f8f304f7ef0>,
E             + ]
test_opensearchpy/test_async/test_connection.py:231: AssertionError
```

**Comment by merlinz01** [comment]
Testing with Python 3.12 works for me, can you rerun the test?

**Comment by dblock** [comment]
> Testing with Python 3.12 works for me, can you rerun the test?

Failed again, it's not a fluke, there's some warning but the contents don't show up in the log (maybe start there to see the error) https://github.com/opensearch-project/opensearch-py/actions/runs/11846183040/job/33061554546?pr=843.

**Comment by merlinz01** [comment]
`enable_cleanup_closed ignored because https://github.com/python/cpython/pull/118960 is fixed in Python version sys.version_info(major=3, minor=12, micro=7, releaselevel='final', serial=0)`

https://github.com/python/cpython/pull/118960

**Comment by merlinz01** [comment]
OK, so it's a deprecated parameter on `aiohttp.TCPConnection` starting in Python 3.12.7, I don't think related to these changes. Shall I add a patch for the test onto this PR?

**Comment by dblock** [comment]
> OK, so it's a deprecated parameter on `aiohttp.TCPConnection` starting in Python 3.12.7, I don't think related to these changes. Shall I add a patch for the test onto this PR?

Yes please, we do want tests to pass :)

**Review by dblock — APPROVED** [review]

**Comment by merlinz01** [comment]
Thanks @dblock! I don't see where the release schedule for clients is documented, so wondering when this change will be available.

**Comment by dblock** [comment]
Open an issue to release v. next and I can try to make it happen this coming week.

**Pull request merged**